### ISSUE TITLE
Publish to github packages

### ISSUE THREAD

**Issue by objorke** [oxyplot/oxyplot#1681]
Move from myget to github packages? 
https://github.com/features/packages

**Comment by VisualMelon** [comment]
Worth exploring even if we don't drop myget, but we will still need AppVeyor for windows builds (worth setting up github actions for the cross-platform stuff?). Are there compelling reasons to switch? I know Avalonia have problems with the number of allowed packages, but we don't have such high through-put.

**Comment by rstm-sf** [comment]
As far as I know, the use of github packages is not an open source way, as it is platform-dependent, otherwise it was not intended

**Comment by Jonarw** [comment]
From my (admittedly fairly limited) understanding it should in fact be possible to do everything we do with AppVeyor with GitHub Actions, including tests on all platforms, pushing 'nightly' builds to GitHub Packages and even pushing (pre-)releases to NuGet.

I found this fairly comprehensive overview over the required steps:
https://dusted.codes/github-actions-for-dotnet-core-nuget-packages

**Comment by objorke** [comment]
I understand it would be easy to use nuget push on a Windows build agent (like we already have on https://github.com/oxyplot/oxyplot/actions). 

I find it compelling to have both the CI build and packages on the same platform, no need to login and handle access separately for appveyor and myget. They could both be depreated if this works.

It would also be interesting to know if GitHub Actions, Packages and Releases support data lineage - e.g. making it possible to verify that the nuget packages are based on the open source code. Do you know?

**Comment by VisualMelon** [comment]
I ran into this recently which may be relevant but I didn't look into it closely: https://github.com/clairernovotny/DeterministicBuilds

**Comment by VisualMelon** [comment]
@rstm-sf could you elaborate on that? I've got a github action building and pushing a package to github packages in another repo, but for the life of me I cannot add it to a project without it requesting authentication (but I can download the nupkg directly from the website without being logged in).

**Comment by rstm-sf** [comment]
Github actions are good in themselves. About github packages:
> You must use a personal access token with the appropriate scopes to publish and install packages in GitHub Packages.
https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-dotnet-cli-for-use-with-github-packages#authenticating-with-a-personal-access-token

**Comment by VisualMelon** [comment]
OK. Unless I'm missing something (I hope I am) then switching to Packages exclusively would seem a bad idea: we shouldn't expect users to authenticate with anything to access pre-releases.

**Comment by objorke** [comment]
Agree, we should not expect users to authenticate for public packages.
https://github.community/t/download-from-github-package-registry-without-authentication/14407/66
